core/modules/ccontrol/UserQuerySelect.cc

```diff
@@ -146,11 +146,12 @@ UserQuerySelect::UserQuerySelect(std::shared_ptr<qproc::QuerySession> const& qs,
                                  std::shared_ptr<qproc::SecondaryIndex> const& secondaryIndex,
                                  std::shared_ptr<qmeta::QMeta> const& queryMetadata,
                                  qmeta::CzarId czarId,
+                                 uint64_t taskId,
                                  std::string const& errorExtra)
     :  _qSession(qs), _messageStore(messageStore), _executive(executive),
        _infileMergerConfig(infileMergerConfig), _secondaryIndex(secondaryIndex),
        _queryMetadata(queryMetadata), _qMetaCzarId(czarId), _qMetaQueryId(0),
-       _killed(false), _submitted(false), _sessionId(0), _sequence(0),
+       _killed(false), _submitted(false), _taskId(taskId), _sequence(0),
```

Comment:
I'm wondering if can come up with another name for _taskId (maybe queryId or userQueryId?). The executive breaks user queries into jobs with jobIds, those are passed to the worker where they are tasks. To avoid confusion, it would be good if they all have unique names.